app/services/sso/openid_connect.rb

```diff
@@ -38,7 +38,12 @@ def bearer_token_set?
 
     def valid_issuer?
       payload = jwt_token.decoded_payload
-      payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+      if payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
```

Comment:
This will fail if `payload` is `nil`.

Reply:
You could use an early return here.
`return true if`

Reply:
I was actually thinking about the other way around: `unless ... log() return false`. If there are more conditions it feels more logical to me.